### DISCUSSION TITLE
When Docker deploys trame as a multi-person application, how to specify different data for different people

### DISCUSSION THREAD

**Discussion by chiyandetaotie** [Kitware/trame#518] [Trame/VTK] [answered]
When running the code locally, I can specify the data address to read in, using the "--data" option. 
The example I use is as follows
![image](https://github.com/Kitware/trame/assets/39592625/9e3463a6-6310-4e39-961f-4850db07f584)
I added the following
![image](https://github.com/Kitware/trame/assets/39592625/4971031b-d9a7-485a-ab94-145d534372c7)
I want to use docker to deploy this application for multiple people to use.
I want to enable different people to use this application and specify different data instead of the same data.
I don't know how to deploy it to achieve this function.

I am using Windows 10,and configured it as follows
![image](https://github.com/Kitware/trame/assets/39592625/44217db2-08d7-426d-8dd7-d58c4b37dc20)
![image](https://github.com/Kitware/trame/assets/39592625/053a5f95-581c-40a5-a005-fcdf78cecfff)
![image](https://github.com/Kitware/trame/assets/39592625/1390763e-08ff-406e-80e9-4af3ae136f91)

How to deploys trame with Docker ?

**Comment by jourdain** [comment]
Just deploy what you showed and just make sure that each user connect with their own URL (`http://localhost:8080/?data=/path/from/within/docker/to/file.vtu`)

**Comment by chiyandetaotie** [comment]
I build the image as follow:
docker build -t trame-multifilter2 .
![image](https://github.com/Kitware/trame/assets/39592625/93e08ce4-f1ef-4f16-99fd-d7b4d6eabe69)
But run the image on port 8080, it show error:
docker run -it --rm -p 8080:80 trame-multifilter2
![image](https://github.com/Kitware/trame/assets/39592625/ea73f625-b4f5-41ef-a9ec-3056e140f331)
How to solve it?

**Comment by jourdain** [comment]
The build step failed but I can't see that on your log. Try using the base image `kitware/trame:py3.9`

**Comment by chiyandetaotie** [comment]
I run the docker as follow:
docker run -it -v D:\webglDemo\data:/mnt --rm -p 8080:80 trame-multifilter2

![image](https://github.com/Kitware/trame/assets/39592625/c9094cfa-8be2-493c-b79d-a3afd0898b84)

I copy the file to the docker (/mnt/disk_out_ref.vtu)

![image](https://github.com/Kitware/trame/assets/39592625/5f0d6176-3218-44ad-8570-e8ff31e4bb1b)

But it has error on the browser when I search "http://localhost:8080/?data=/mnt/disk_out_ref.vtu"
![image](https://github.com/Kitware/trame/assets/39592625/f82c8c2d-2053-4ff8-a753-afb8bb6b1faf)

Also shows error in terminal as follow:
![image](https://github.com/Kitware/trame/assets/39592625/8ed2e583-ba80-4a09-8ae4-b03084c33d43)

**Comment by jourdain** [comment]
See that [comment](https://github.com/Kitware/trame/discussions/520#discussioncomment-9391319) to figure out what log to look at to figure out the issue with your app.

**Reply by chiyandetaotie** [reply, marked as answer]
Thanks. I have found the reason and changed it. 
I used the wrong method. like this:
![image](https://github.com/Kitware/trame/assets/39592625/b577ec0f-1069-4de3-b3bc-d493ce22f5b3)
 And I change it as follow:
![image](https://github.com/Kitware/trame/assets/39592625/1c4e4f68-6b59-4223-b18f-b8ebf274b6e0)
Now it works:
![image](https://github.com/Kitware/trame/assets/39592625/37e7a5d2-e263-4b26-9afc-ad5652050de8)
